src/nanoemoji/write_variable_font.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Assembles UFOs into a variable font.
+
+Based on https://gist.github.com/anthrotype/c11515065a4aa6549f9d5f2dfdcf8f23"""
+
+from absl import app
+from fontTools import designspaceLib
+from nanoemoji import config
+import ufo2ft
+import ufoLib2
+
+def only(filter_fn, iterable):
+    it = filter(filter_fn, iterable)
+    result = next(it)
+    assert next(it, None) is None
+    return result
+
+
+def main(argv):
+    ufos = tuple(filter(lambda a: a.endswith(".ufo"), argv[1:]))
+    #output_file = only(lambda a: not a.endswith(".ufo"), argv[1:])
+
+    font_config = config.load()
+
+    designspace = designspaceLib.DesignSpaceDocument()
+
+    import pprint
+    pp = pprint.PrettyPrinter()
+
+    # define axes names, tags and min/default/max
+    axis_defs = [
+        dict(
+            tag=a.axisTag,
+            name=a.name,
+            minimum=min(p.position for m in font_config.masters
+                        for p in m.position
+                        if p.axisTag == a.axisTag),
+            default=a.default,
+            maximum=max(p.position for m in font_config.masters
+                        for p in m.position
+                        if p.axisTag == a.axisTag),
+        ) for a in font_config.axes
+    ]
+    pp.pprint(axis_defs)
+    for axis_def in axis_defs:
+        designspace.addAxisDescriptor(**axis_def)
+
+    axis_names = {a.axisTag: a.name for a in font_config.axes}
+    for master in font_config.masters:
+        ufo = ufoLib2.Font.open(master.output_ufo)
+        ufo.info.styleName = master.style_name
+        location = {axis_names[p.axisTag]: p.position for p in master.position}
+        designspace.addSourceDescriptor(name=master.output_ufo, location=location, font=ufo)
+
+    # build a variable TTFont from the designspace document
+    # TODO what if I wanted a variable CFF?!
```

Comment:
if you want a variable CFF2 font you do `ufo2ft.compileVariableCFF2`
https://github.com/googlefonts/ufo2ft/blob/e1c08f6efb07a2fb5202d65a90af94dbd599e7e8/Lib/ufo2ft/__init__.py#L617

Reply:
Is there / could there be a compileVariableCFF?

Reply:
no, only CFF2 can be variable

Reply:
I suggest changing the comment here to read "TODO: Use ufo2ft.compileVariableCFF2`

Reply:
Done